test/e2e/network/dns.go

```diff
@@ -642,4 +644,50 @@ var _ = common.SIGDescribe("DNS HostNetwork", func() {
 		validateDNSResults(ctx, f, pod, append(wheezyFileNames, jessieFileNames...))
 	})
 
+	ginkgo.It("should match hostname for a Pod with container hostname", func(ctx context.Context) {
+		ginkgo.By("Creating a pod by setting a hostname")
+
+		testAgnhostPod := e2epod.NewAgnhostPod(f.Namespace.Name, "test-dns-hostname", nil, nil, nil)
+		testAgnhostPod.Spec.Hostname = dnsTestPodHostName
+		testAgnhostPod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, testAgnhostPod, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to created pod: %s", testAgnhostPod.Name)
+
+		err = e2epod.WaitTimeoutForPodReadyInNamespace(ctx, f.ClientSet, testAgnhostPod.Name, f.Namespace.Name, framework.PodStartTimeout)
+		framework.ExpectNoError(err, "failed to wait for pod %s to be running", testAgnhostPod.Name)
+
+		stdout, err := e2eoutput.RunHostCmd(testAgnhostPod.Namespace, testAgnhostPod.Name, "hostname")
+		framework.ExpectNoError(err, "failed to run command hostname: %s", stdout)
+		trimmed := strings.TrimSpace(stdout)
+		if testAgnhostPod.Spec.Hostname != trimmed {
+			framework.Failf("expected hostname: %s, got: %s", testAgnhostPod.Spec.Hostname, trimmed)
+		}
+	})
+
+	ginkgo.It("should match container hostname with targetNode when hostNetwork is set to true for a Pod", func(ctx context.Context) {
```

Comment:
we might need to work on the test names.

Reply:
How about merging these two tests together and
"should ignore the specified hostname when the host network is set to true"  and as test name?

Reply:
not merging, both are different use cases , if one fails and the other no it will be confusing to debug
```suggestion
        // https://issues.k8s.io/67019 
	ginkgo.It("spec.Hostname field is silently ignored and the node hostname is used when hostNetwork is set to true for a Pod", func(ctx context.Context) {
```